desktop/core/src/desktop/js/reactComponents/FileUploadQueue/FileUploadQueue.tsx

```diff
@@ -66,28 +86,98 @@ const FileUploadQueue = (): JSX.Element => {
     onComplete
   });
 
+  // If there are already files in the queue (e.g., restored state), ensure the queue is visible
+  useEffect(() => {
+    if (uploadQueue.length > 0) {
+      setIsVisible(true);
+    }
+  }, [uploadQueue.length]);
+
   useHuePubSub<FileUploadEvent>({
     topic: FILE_UPLOAD_START_EVENT,
-    callback: (data?: FileUploadEvent) => {
-      if (data?.files) {
-        setIsVisible(true);
-        addFiles(data.files);
+    callback: async (data?: FileUploadEvent) => {
+      const newFiles = data?.files ?? [];
+      if (newFiles.length > 0) {
+        const { conflicts, nonConflictingFiles } = await resolveFileConflicts(
+          newFiles,
+          uploadQueue
+        );
+        if (conflicts.length > 0) {
+          setConflictingFiles(conflicts);
+        } else {
+          setConflictingFiles([]);
+        }
+        if (nonConflictingFiles.length > 0) {
+          addFiles(nonConflictingFiles);
+          setIsVisible(true);
+        }
       }
     }
   });
 
+  const resolveFileConflicts = async (
```

Comment:
Is this name correct? It is not actually resolving the conflicts right, it only detects them....